app/models/user.rb

```diff
@@ -259,7 +259,7 @@ class User < ApplicationRecord
   before_validation :set_username
   before_validation :strip_payment_pointer
   before_create :set_default_language
-  before_destroy :unsubscribe_from_newsletters, prepend: true
+  before_destroy :unsubscribe_from_newsletters, prepend: true, if: -> { SiteConfig.mailchimp_newsletter_enabled? }
```

Comment:
It seems this is the only place we need this method. I wonder if you want to remove it again from `SiteConfig` and instead update `unsubscribe_from_newsletters` with a guard clause:

```ruby
def unsubscribe_from_newsletters
  return unless SiteConfig.mailchimp_api_key.present? && SiteConfig.mailchimp_newsletter_id.present?
end
```

I'm not actually sure if that's better, but at least it's more self-contained.